### ISSUE TITLE
Error while opening custom ODK collect after selecting a feature and clicked on 'Go to ODK'

### ISSUE THREAD

**Issue by manjitapandey** [hotosm/fmtm#1780]
**Describe the bug**
An error is occured when go to odk is clicked after selecting a feature on fmtm mobile. 

**To Reproduce**

1. Open fmtm on mobile
2. select any feature and click on go to odk.

**Expected behavior**
The ODK should open with prefilled fields, when go to odk is clicked

**Screenshots**

https://github.com/user-attachments/assets/9c8cd716-c354-4e5e-bd6c-b58f9617d792

**Comment by spwoodcock** [comment]
'Go to ODK' will open with the task id prefilled.

If you click on the building, then the button 'Map Feature in ODK', does this work loading the entity id?

**Comment by manjitapandey** [comment]
Yes, clicking on building 'go to odk' feature work well. but its on task level.

**Comment by spwoodcock** [comment]
Fixed in latest build 😄

https://github.com/hotosm/odkcollect/releases/tag/v2024.2.4-entity-select

I will link this in the sidebar for FMTM as the default download soon

**Comment by spwoodcock** [comment]
Please test the APK @manjitapandey and close if it works for you 👍

**Comment by manjitapandey** [comment]
Sure, is the updated apk linked on sidebar of FMTM??

**Comment by manjitapandey** [comment]
The custom odk keeps crashing on my device. 
and also when I go to odk by locking a task, it redirect me to a section to scan the QR instead of form. 
when I go to odk selecting an entity, it just crashes or show the error message as in image below. I also tried deleting the cache and overall staging data. 

Uploading Record_2024-09-13-11-19-11.mp4…

![Screenshot_2024-09-13-11-11-07-40_171725cd2e369f62c287134214eaed67](https://github.com/user-attachments/assets/0fb3392a-2b25-436c-8c8b-b79cfabba8c3)

**Comment by spwoodcock** [comment]
Can you retry this?

People were using the app in the field it seems, so it must work

**Comment by spwoodcock** [comment]
Nice! The feature is actually pre-selected if you skip the task_id choice, but choosing a task id causes the feature to deselect!

First of all, the task_id should be prefilled when the form is opened - this is a bug.

Next, the XLSForm needs logic to:
- Hide the task_id question if feature is selected already (possible, but this might complicate things)
- Hide the 'do you want to add a new geometry' question if a feature is pre-selected.

**Comment by spwoodcock** [comment]
Significantly updated the XLSForm logic - please test all scenarios - they are working for me now 😄 

- New feature upload (automatically marks the geom as digitised correctly / read-only field)
- Select geom, mark as not existing, exit form.
- Select geom, mark as existing, fill form, mark as not digitised correctly, select why
- Select geom, mark as existing, fill form, mark as digitised correctly

**Comment by manjitapandey** [comment]
There is configuration message. However, I am using the custom ODK and i have ODK open on my mobile. 
![Screenshot_2024-09-24-13-23-14-05_171725cd2e369f62c287134214eaed67](https://github.com/user-attachments/assets/14f5a90b-1e03-42f6-969e-dc2cd7b6ddb0)
![Screenshot_2024-09-24-13-34-29-72_171725cd2e369f62c287134214eaed67](https://github.com/user-attachments/assets/2c022dcf-be20-4feb-917c-0635c4e1d3f3)


Is configuring the project beforehand the expected behavior?

**Comment by spwoodcock** [comment]
Yep!

There's needs to be a form loaded for it to be opened and pre-filled by intent 😄
